src/flows/context/query.tsx

```diff
@@ -106,10 +110,13 @@ export const QueryProvider: FC<Props> = ({children, variables, org}) => {
   const query = (text: string) => {
     const windowVars = getWindowVars(text, vars)
     const extern = buildVarsOption([...vars, ...windowVars])
+    const queryID = generateHashedQueryID(text, variables, org.id)
 
     event('runQuery', {context: 'flows'})
-    return runQuery(org.id, text, extern)
-      .promise.then(raw => {
+    const result = runQuery(org.id, text, extern)
```

Comment:
not feature flagging here since the entire feature is flagged

Reply:
flagged by the same flag?

Reply:
It's flagged under the flows feature flag, which basically means we're enabling the feature for anyone who has access to flows

Reply:
that's not the right way of thinking about flags. we shouldn't have to turn off flows to disable the side effects of the cancel button changing. they should be able to be toggled independently from each other, not enabled / disabled by convention

Reply:
k